### PR TITLE
add check.ignore to list cargo check diagnostics to ignore (dead_code, unused_imports, ...)

### LINKED ISSUE #14798: disable/filter cargo check diagnostics
The `unused_code` diagnostic drives me crazy when writing something new, too many useless annotations.

I made a patch to filter `cargo check` diagnostics here: https://github.com/adamse/rust-analyzer/commit/42a52ce2f8e50a395070dab885ac566e57b955a4
(I've extended the existing `rust-analyzer.diagnostics.disabled`  setting to also apply to `cargo check`.)

If the project is interested in this feature I would be happy to upstream it.

### PR DESCRIPTION
**Pull request by adamse**
fixes #14798

### CODE REVIEW

File: crates/rust-analyzer/src/config.rs [hunk 1]
```diff
@@ -150,6 +150,10 @@ config_data! {
         ///
         /// Set to `"all"` to pass `--all-features` to Cargo.
         check_features | checkOnSave_features: Option<CargoFeaturesDef>  = "null",
+        /// List of check diagnostics to ignore.
```
Comment (HKalbasi) [inline crates/rust-analyzer/src/config.rs:153]: Can you clarify it a bit more? For example: List of `cargo check` (or other command specified in `check.command`) diagnostics to ignore.

Comment (lnicola) [reply crates/rust-analyzer/src/config.rs:153]: Gentle ping @adamse

Comment (adamse) [reply crates/rust-analyzer/src/config.rs:153]: thank you. i’ll update in a week or so!

**Comment by HKalbasi** [comment]
Thanks!
@bors r+

**Pull request merged**